doc/user/methods/random_forest_regressor.md

```diff
@@ -0,0 +1,639 @@
+with the performance of each individual tree:
+
+```c++
+// Load a categorical dataset.
+arma::mat data;
+mlpack::data::DatasetInfo info;
+// See https://datasets.mlpack.org/telecom_churn.arff.
+mlpack::data::Load("telecom_churn.arff", data, info, true);
+
+arma::rowvec responses;
+// See https://datasets.mlpack.org/telecom_churn.responses.csv.
+mlpack::data::Load("telecom_churn.responses.csv", responses, true);
+
+// Split data into training set (80%) and test set (20%).
+arma::mat trainData, testData;
+arma::rowvec trainResponses, testResponses;
+mlpack::data::Split(data, responses, trainData, testData, trainResponses,
+    testResponses, 0.2);
+
+// Create the tree.
+mlpack::RandomForestRegressor rf;
+// Train on the given dataset, specifying a minimum gain of 1e-10 and keeping the
+// default minimum leaf size.
+const double mse = rf.Train(trainData, info, trainResponses, 20 /*number of trees*/
+    ,1 /* minimum leaf size */, 1e-10 /* minimum gain */);
+
+// Compute test accuracy for each tree
+arma::rowvec testPredictions;
+for(size_t i = 0; i<rf.NumTrees(); i++)
+{
+  rf.Tree(i).Predict(testData, testPredictions);
+  const double testAverageError = arma::mean(testResponses - testPredictions);
+  std::cout << "Tree " << i << " has averageError: " << testAverageError << "."
+    << std::endl;
+}
+
+// Compute predictions on test data using whole forest.
+rf.Predict(testData, testPredictions);
+
+// Compute the average error on the test set.
+const double testAverageError = arma::mean(testResponses - testPredictions);
+std::cout << "The whole forest has Average error: " << testAverageError << "."
+    << std::endl;
+```
+
+### Advanced Functionality: Template Parameters
+
+#### Using different element types.
+
+`RandomForestRegressor`'s contructors, `Train()`, and `Predict()` functions support 
+any data type, so long as it supports the Armadillo matrix API. So, for instance, learning can be done on single-precision floating-point data:
+
+```c++
+// 1000 random points in 10 dimensions.
+arma::fmat dataset(10, 1000, arma::fill::randu);
+// Random responses for each point, with a normal distribution.
+arma::frowvec responses = arma::randn<arma::frowvec>(1000);
+
+// Train in the constructor.
+mlpack::RandomForestRegressor rf(dataset, responses);
+
+// Create test data (500 points).
+arma::fmat testDataset(10, 500, arma::fill::randu);
+arma::frowvec predictions;
+rf.Predict(testDataset, predictions);
+// Now `predictions` holds predictions for the test dataset.
+
+// Print some information about the test predictions.
+std::cout << arma::accu(predictions > 1) << " test points predicted to have "
+    << "value greater than 1." << std::endl;
+```
+
+---
+
+#### Fully custom behavior.
+
+The `RandomForestRegressor` class also supports several template parameters, 
+which can be used for custom behavior during learning. The full signature of 
+the class is as follow:
+
+```c++
```

Comment:
I can't provide a suggestion here, but try removing the `c++` here.  Otherwise the `test-docs.sh` script will try and build this as a standalone code snippet.  So I have been, in other documentation, just leaving it as an unspecified type code block (and the formatting looks fine).

I know that is a little bit of "hidden knowledge", I do plan to document it (I just haven't done it yet).